src/Runner.Worker/Handlers/StepHost.cs

```diff
@@ -239,5 +241,27 @@ public async Task<string> DetermineNodeRuntimeVersion(IExecutionContext executio
                                                          cancellationToken: cancellationToken);
             }
         }
+
+        public DictionaryContextData GetExpressionValues(IExecutionContext executionContext)
+        {
+            var expressionValues = executionContext.ExpressionValues.Clone() as DictionaryContextData;
+            UpdatePathsInExpressionValues("github", expressionValues);
+            UpdatePathsInExpressionValues("runner", expressionValues);
+            return expressionValues;
+        }
+
+        private void UpdatePathsInExpressionValues(string contextName, DictionaryContextData expressionValues) 
+        {
+            var dict = expressionValues[contextName].AssertDictionary($"expected context {contextName} to be a dictionary");
+            foreach (var key in dict.Keys.ToList())
+            {
+                var value = dict[key]?.ToString();
+                if (!string.IsNullOrEmpty(value))
```

Comment:
We need to only modify it if it is a string context data, otherwise we are going to blow up the `github.event` which is an object